common/lib/Warewulf/SystemFactory.pm

```diff
@@ -51,8 +51,21 @@ new($$)
             $type = "rhel";
         } elsif (-f "/etc/SuSE-release") {
             $type = "Suse";
+        } elsif ( -f "/etc/os-release") {
+            open(SYSTEM, "/etc/os-release\n");
+            while (<SYSTEM>) {
+                if ($_ =~ /NAME=/) {
+                    if ($_ =~ /SUSE|SLE/) {
+                        $type = "Suse";
+                    }
+                    last;
+                }
+            }
+            close SYSTEM;
+            goto NOSYS if (! $type);
```

Comment:
If you could help clean up our existing code a bit, instead of using a `goto` to get into the catch all `else`, add a conditional before `} elsif ( -f "/etc/os-release") {` to check for `/etc/debian_version` and set $type = "Deb". Then under the catch all `else` set type to `unsupported`.

Reply:
Fixed now.